### ISSUE TITLE
Improve identity parsing

### ISSUE THREAD

**Issue by alexhauser** [sqrldev/SQRLDotNetClient#1]
Hi Jose,

I've noticed that `SQRL.ImportSqrlIdentityFromFile(string file)` currently expects an identity file to have all 3 block types present. This is problematic, since most of the identities "in the wild" will not contain a block of type 3, since it will only present if an identity has been rekeyed at least once.

Also, the current parsing code expects the blocks to be present in a specific order.

In general, I would suggest **not to assume anything** in terms of which block types are present, and also not in which order they are present. 

My suggestion would therefore be to do the parsing like this:

(Pseudocode)
```pseudocode
i=0
while (i<file_len)
{
    if (i+4 < file_len) break

    blockLen = id_data.skip(0).take(2)
    blockType  = id_data.skip(i+2).take(2)

    switch(blockType)
        case 1:
        parseBlock1();

    case 2:
        parseBlock2();
    .
    .
    .
    default:
        parseUnknownBlock();

    i += blockLen     
}
```

Also, a SQRL client should handle unknown block types gracefully, meaning they should be ignored but retained. The current implementation does not handle that at all.

So, I would suggest to create a "standard" SQRL block for holding unknown block types, something like:

```C#
public class SQRLBlock : ISQRLBlock
    {
        public ushort Length { get; } = 125;
        public ushort Type { get; } = 1;

        public byte[] BlockData { get; set; }

        public void FromByteArray(byte[] blockData)  { ... }
        public byte[] ToByteArray() { ... }
}
```
And lastly, you could think about replacing the fixed block properties within `SQRLIdentity`

```C#
    public SQRLBlock1 Block1 { get; set; }
    public SQRLBlock2 Block2 { get; set; }
    public SQRLBlock3 Block3{ get; set; }
```
with something like ...
```C#
    public List<ISQRLBlock> Blocks { get;}
```
... and add convenience methods like `SQRLIdentity.HasBlock(ushort type)` and `ISQRLBlock SQRLIdentity.GetBlock(ushort type)`.

The advantage of this approach is that you can retain unknown blocks and at the same time retain the block order of the source identity file.

Thoughts?

Cheers, Alex

**Comment by alexhauser** [comment]
Oh, and also, `SQRL.ImportSqrlIdentityFromFile(string file)` could be replaced by a public static method `SQRLIdentity.FromFile(String file)`, which would internally call another public static method `SQRLIdentity.FromByteArray(byte[] identityData)`. This would make it very flexible for the consumer of the library and is probably more intuitive than putting it in a utility function.

**Comment by josegomez** [comment]
Thanks Alex,
This sounds good, my only qualm is I hate "magic strings" so I prefer to have a fixed class defined block types at least for the 3 block types we know.
We could use inheritance for this, so a "Generic" BlockType but then have Block1 and Block2 and Block3 inherit from there so that we could have fixed properties for these blocks?

I'll go look at your PR see how you approached it.

**Comment by josegomez** [comment]
Looks like you already address both of my "issues" with your approach, awesome job! Let me know when you are ready for merging the PR!

Thank you!

**Comment by alexhauser** [comment]
Yeah, seems like we're on the same page when it comes to architectural questions. Nice!

**Comment by josegomez** [comment]
Fixed by PR #2
